File: SYP.Web/Modules/ServerTypes/Email/EmailTemplatesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmailTemplatesRow } from "./EmailTemplatesRow";

export interface EmailTemplatesColumns {
    Id: Column<EmailTemplatesRow>;
    TemplateKey: Column<EmailTemplatesRow>;
    Name: Column<EmailTemplatesRow>;
    Subject: Column<EmailTemplatesRow>;
    Category: Column<EmailTemplatesRow>;
    LanguageId: Column<EmailTemplatesRow>;
    IsActive: Column<EmailTemplatesRow>;
}

export class EmailTemplatesColumns extends ColumnsBase<EmailTemplatesRow> {
    static readonly columnsKey = 'Email.EmailTemplates';
    static readonly Fields = fieldsProxy<EmailTemplatesColumns>();
}

File: SYP.Web/Modules/Email/EmailTemplates/EmailTemplatesColumns.cs
namespace SYP.Email.Columns;

[ColumnsScript("Email.EmailTemplates")]
[BasedOnRow(typeof(EmailTemplatesRow), CheckNames = true)]
public class EmailTemplatesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public int Id { get; set; }

    [EditLink, Width(150)]
    public string TemplateKey { get; set; }

    [Width(200)]
    public string Name { get; set; }

    [Width(250)]
    public string Subject { get; set; }

    [Width(100)]
    public string Category { get; set; }

    [Width(80)]
    public string LanguageId { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}
